dronegen/relcli.go

```diff
@@ -31,28 +31,39 @@ func relcliPipeline(trigger trigger, name string, stepName string, command strin
 			},
 		},
 		waitForDockerStep(),
-		pullRelcliStep(),
+		kubernetesAssumeAwsRoleStep(kubernetesRoleSettings{
+			awsRoleSettings: awsRoleSettings{
+				awsAccessKeyId:     value{fromSecret: "TELEPORT_BUILD_USER_READ_ONLY_KEY"},
+				awsSecretAccessKey: value{fromSecret: "TELEPORT_BUILD_USER_READ_ONLY_SECRET"},
+				role:               value{fromSecret: "TELEPORT_BUILD_READ_ONLY_AWS_ROLE"},
+			},
+			configVolume: volumeRefAwsConfig,
+		}),
+		pullRelcliStep(volumeRefAwsConfig),
 		executeRelcliStep(stepName, command),
 	}
 
-	p.Services = []service{
-		dockerService(volumeRefTmpfs),
+	p.Services = []service{dockerService(volumeRefTmpfs)}
+	p.Volumes = []volume{
```

Comment:
Seems like `dockerVolumes(volumeTmpfs, volumeAwsConfig)` would be equivalent.

Reply:
It is equivalent.  However while I was looking at this pattern I didn't understand what the purpose of `dockerVolumes` over `[]volume{dockerVolume}` was.  It seems like an unnecessary abstraction that makes docker volumes handled in a different manner than all other volumes (which don't have functions afaict).  Perhaps you can elucidate on why we'd prefer one or the other?

I'll readily admit, having two patterns in the repo for the same thing is worse than having one.

Reply:
Removed `dockerVolumes` and `dockerVolumeRefs` in https://github.com/gravitational/teleport/pull/17201/commits/0f21054099ec681677169d0c685a59eacbfaf6cd.  Why did I go this route instead of vice versa?

 * We don't use a similar pattern with other volume refs
 * This removes the need to look in another file to determine that a single ref is added
 * Adding the docker volumes in an inline struct is fewer lines of code

Let me know if you disagree.